module/os/linux/zfs/zfs_uio.c

```diff
@@ -507,12 +504,13 @@ static int
 zfs_uio_pin_user_pages(zfs_uio_t *uio, zfs_uio_rw_t rw)
 {
 	long res;
-	size_t skip = uio->uio_skip;
-	size_t len = uio->uio_resid - skip;
+	size_t len = uio->uio_resid;
```

Comment:
Looking again, I think this function should get `size_t skip = uio->uio_iter->iov_offset` instead of removing it, similar to what iov_iter_alignment() and iov_iter_alignment_iovec() do in Linux kernel.  It was wrong to call `iov_iter_advance()` on the `iov_offset` value, when we are getting inside the `uio_iter`, we should take it into account.

Reply:
@robn I agree with @amotin on this one. So this code was me having to do things by hand outside of `iov_iter` interfaces because they are GPL only. We would in fact need to take into account the `iov_offset` here.